### PR TITLE
Starting to eliminate remaining usages of old javadoc API and impleme…

### PR DESCRIPTION
**Pull request by jlahoda**
…ntation.

### CODE REVIEW

**Comment by jlahoda** [comment]
FWIW, review for a patch that actually removes the old javadoc API from JDK:
https://mail.openjdk.java.net/pipermail/javadoc-dev/2019-February/000848.html

**Comment by arusinha** [comment]
Hi @jlahoda . Request you to merge  [PR-21](https://github.com/jlahoda/incubator-netbeans/pull/21)
In the pull request I had incorporated your patch for nb-javac for removing support of old javadoc APIs[]

**Comment by jlahoda** [comment]
I think it would be good to merge this - Geertjan, others, could you please take a look? Thanks!

File: nbbuild/templates/projectized.xml [hunk 1]
```diff
@@ -112,6 +112,9 @@ If you are sure you want to build with JDK 9+ anyway, use: -Dpermit.jdk9.builds=
                       compilecp="test.qa-functional.compile.cp"
                       compiledep="test.qa-functional.testdep"/>
         </parseprojectxml>
+        <condition property="requires.nb.javac" value="true">
+            <contains string="${module.classpath}" substring="/org-netbeans-libs-javacapi.jar" />
+        </condition>
```
Comment (rtaneja1) [inline nbbuild/templates/projectized.xml]: <condition property="bootclasspath.prepend" value="${bootclasspath.prepend.vanilla}">
            <istrue value="${requires.nb.javac}"/>
        </condition>
Can you please confirm if this is correct.

File: java/java.source.base/src/org/netbeans/modules/java/source/parsing/JavacParser.java [hunk 2]
```diff
@@ -1333,7 +1326,7 @@ public void setProcessors(Collection<? extends Processor> processors) {
         @Override
         public void enhance(Context context, boolean backgroundCompilation) {
             if (!backgroundCompilation)
```
Comment (rtaneja1) [inline java/java.source.base/src/org/netbeans/modules/java/source/parsing/JavacParser.java]: Can if check be removed?

File: java/java.source.base/src/org/netbeans/modules/java/source/parsing/JavacParser.java [hunk 3]
```diff
@@ -85,7 +85,7 @@
 import org.netbeans.modules.java.preprocessorbridge.spi.JavaFileFilterImplementation;
 import org.netbeans.modules.java.source.JavaFileFilterQuery;
 import org.netbeans.modules.java.source.JavaSourceAccessor;
-import org.netbeans.modules.java.source.JavadocEnv;
+//import org.netbeans.modules.java.source.JavadocEnv;
```
Comment (rtaneja1) [inline java/java.source.base/src/org/netbeans/modules/java/source/parsing/JavacParser.java]: remove this?

File: nbbuild/templates/projectized.xml (hunk 1 above)
Comment (jlahoda) [reply nbbuild/templates/projectized.xml]: The intent here is to make requires.nb.javac inferred automatically (for nb.org modules only, I think, as projectized.xml contains enhancements for nb.org modules). If a module depends on javacapi, then it should get the proper version of the API automatically (by setting the property). Not strictly necessary for this version of this patch, but feels generally useful and was needed during development.

File: java/java.source.base/src/org/netbeans/modules/java/source/parsing/JavacParser.java [hunk 4]
```diff
@@ -1333,7 +1326,7 @@ public void setProcessors(Collection<? extends Processor> processors) {
         @Override
         public void enhance(Context context, boolean backgroundCompilation) {
             if (!backgroundCompilation)
-                NBJavadocClassFinder.preRegister(context);
```
[the remark this answers is not in the document]
Comment (jlahoda) [reply java/java.source.base/src/org/netbeans/modules/java/source/parsing/JavacParser.java]: Done.

[the remark this answers is not in the document]
Comment (jlahoda) [reply java/java.source.base/src/org/netbeans/modules/java/source/parsing/JavacParser.java]: Thanks, done.

File: java/java.source.base/src/org/netbeans/modules/java/source/parsing/JavacParser.java (hunk 2 above)
Comment (jlahoda) [reply java/java.source.base/src/org/netbeans/modules/java/source/parsing/JavacParser.java]: Thanks, done.

File: java/java.source.base/src/org/netbeans/modules/java/source/parsing/JavacParser.java (hunk 3 above)
Comment (jlahoda) [reply java/java.source.base/src/org/netbeans/modules/java/source/parsing/JavacParser.java]: Thanks, done.

**Comment by GeertjanWielenga** [comment]
Great!

**Pull request merged**